### DISCUSSION TITLE
Inquiry about Performance in Larger Projects using twrnc

### DISCUSSION THREAD

**Discussion by VitorAndrey** [jaredh159/tailwind-react-native-classnames#265] [Q&A]
Hello twrnc Team,

I hope this message finds you well. I am currently exploring options for a new UI library for Tailwind in my React Native project, which is expected to be highly scalable. After reviewing your library, I'm impressed with its features and ease of use.

However, as performance is a critical factor for the success of our app, I would like to kindly inquire about the library's performance in larger projects. Specifically, I am interested in understanding how twrnc performs in projects with a substantial codebase.

I understand that each project is unique, but any insights you can provide regarding performance optimization or comparisons to the default React Native StyleSheet would be greatly appreciated. Additionally, if there are any best practices or guidelines you recommend for ensuring optimal performance in larger applications, I would be eager to learn about them.

I genuinely value the work you have put into developing twrnc, and I believe that your insights will greatly assist me in making an informed decision for our project.

Thank you for your time and consideration. I look forward to hearing from you.

Best regards,
Vitor

**Comment by jaredh159** [comment]
i don't have any super clear answers for you. i can tell you the library is heavily optimized with a couple layers of caching, and strives to be very performant.  there is an independent benchmark showing `twrnc` as the fastest of the tailwind+rn libraries, but i'm not sure how much stock to put into that, see [this comment](https://github.com/jaredh159/tailwind-react-native-classnames/discussions/171#discussioncomment-7897651). i can tell you that performance has not been a problem for 2 apps i have in the app store that use it, and i've not heard any real-world reports of the library being a perf problem. but there's really no way i can guarantee it will fit your use case and perform under the conditions you might encounter.

**Reply by VitorAndrey** [reply]
Thank you for your time and assistance.. It's reassuring to hear about your positive experiences with the library in your own apps!
